pkgs/by-name/dp/dprint/plugins/update-plugins.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i python -p nix 'python3.withPackages (pp: with pp; [ requests ])'
+
+import json
+import subprocess
+import requests
+
+
+def nix_prefetch_url(url, algo="sha256"):
+    hash = (
+        subprocess.check_output(["nix-prefetch-url", "--type", algo, url])
+        .decode("utf-8")
+        .rstrip()
+    )
+    sri = (
+        subprocess.check_output(
+            ["nix", "hash", "convert", "--hash-algo", algo, "--to", "sri", hash]
+        )
+        .decode("utf-8")
+        .rstrip()
+    )
+    return sri
+
+
+def update_plugins():
+    data = requests.get("https://plugins.dprint.dev/info.json").json()["latest"]
```

Comment:
I noticed the JSON file also has interesting data for file extensions and excludes that could be re-used in treefmt-nix.

It's possible to preserve those keys in the passthru argument to fetchurl.

Reply:
I will add this as well in the code generation script.

Reply:
added these as `dprint-plugins.<plugin>.passthru.initConfig`, hash the whole plugins.nix attributes per each respective <plugin>.

Reply:
Nice work, it's taking a good shape.

In terms of Nix style, I would push the common setup into a `mkDprintPlugin` constructor. What's good about a constructor is that it can be re-exposed to the user and provide them with more flexibility. This is useful for cases where they want to build their own plugins from different sources or versions.

You can also split each plugin into their own files and then use a combination of `builtins.readDir` to load each of the plugins separately. What's nice about this approach is that conflict merges are less likely to happen.

Putting all of this together, it would look a bit like this:

`default.nix`
```nix
{ lib, fetchurl }:
let
  mkDprintPlugin = { url, hash, pname, version, description, initConfig }:
    fetchurl {
      inherit hash url;
      name = "${pname}-${version}.wasm";
      meta.description = description;
      passthru.updateScript = ./update-plugins.py;
      passthru.initConfig = initConfig;
    };
  
  plugins = lib.mapAttrs (name: _: import name { inherit mkDprintPlugin }) (lib.filterAttrs (name: type: type == "regular" && name != "default.nix" && lib.hasSuffix ".nix" name) (builtins.readDir ./.));
in
plugins // { inherit mkDprintPlugin; }
```

And then each plugin file is:
```nix
{ mkDprintPlugin }:
mkDprintPlugin {
  pname = "g-plane-markup_fmt";
  version = "0.18.0";
  description = "HTML, Vue, Svelte, Astro, Angular, Jinja, Twig, Nunjucks, and Vento formatter.";
  url = "https://plugins.dprint.dev/g-plane/markup_fmt-v0.18.0.wasm";
  hash = "sha256-G8UnJbc+oZ60V3oi8W2SS6H06zEYfY3wpmSUp+1GF8k=";
  updateUrl = "https://plugins.dprint.dev/g-plane/markup_fmt/latest.json";
  initConfig = {
    configExcludes = [ ];
    configKey = "markup";
    fileExtensions = [
      "html"
      "vue"
      "svelte"
      "astro"
      "jinja"
      "jinja2"
      "twig"
      "njk"
      "vto"
    ];
  };
}
```

Reply:
I did it, but this doesn't allow using `nix-shell maintainers/scripts/update.nix --argstr skip-prompt true --argstr path dprint-plugins`.

Reply:
`import name` I tried `import ./${name}` but getting a nixpkgs-vet error. it does work locally on my machine.